scripts/add_renewal_events.py

```diff
@@ -0,0 +1,46 @@
+import sys
+import uuid
+
+import os
+
+root_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))
+sys.path.insert(0, root_dir)
+os.environ["DJANGO_SETTINGS_MODULE"] = "atmosphere.settings"
+
+import django
+
+django.setup()
+from jetstream.models import TASAPIDriver
+from core.models import EventTable, AllocationSource
+
+# fetch values from TACC
+api = TASAPIDriver()
+allocations = api.get_all_projects()
+
+for source in allocations:
+    for details in source['allocations']:
+
+        hashstring = '%s_%s' % (source['chargeCode'], details['id'])
+        hashed_uuid = uuid.uuid5(uuid.NAMESPACE_DNS, str(hashstring))
+
+        payload = {
+            "start_date": details['start'],
+            "end_date": details['end'],
+            "compute_allocated": details['computeAllocated'],
+            "allocation_source_name": source['chargeCode'],
+        }
+
+        try:
+            aso = AllocationSource.objects.filter(name=payload["allocation_source_name"])
+            if not aso:
+                continue
+
+            e = EventTable(
+                uuid=hashed_uuid,
+                name="allocation_source_created_or_renewed",
+                payload=payload,
+                entity_id=payload['allocation_source_name'],
+                timestamp=payload['start_date'])
+            e.save()
+        except:
+            continue
```

Comment:
So this script adds create_or_renewal event for all the existing allocation sources (update_snapshot in jetstream/tasks.py looks at this event for a start date before computing user_allocation_snapshots)

 if an event already exists then we will get an error because of the unique UUID constraint. This would also mean we do not need to create an event  for the allocation source at all. 
But this script is not in use anymore and can be deleted. This was converted to migration 0079 yesterday